include/networkit/matching/BSuitorMatcher.hpp

```diff
@@ -0,0 +1,175 @@
+        } else {
+            auto ret = min;
+            remove(min.id);
+            return ret;
+        }
+    }
+
+    void insert(const Node &u) {
+        assert(partners.size() < max_size);
+        partners.emplace_back(u);
+        if (partners.size() == max_size && !partners.empty()) {
+            min = *std::min_element(partners.begin(), partners.end(),
+                                    [](const Node &x, const Node &y) {
+                                        if (x.weight == y.weight) {
+                                            return x.id > y.id;
+                                        }
+                                        return x.weight < y.weight;
+                                    });
+        }
+    }
+
+    void remove(node u) {
+        partners.erase(std::remove_if(partners.begin(), partners.end(),
+                                      [u](const Node &v) {
+                                          if (v.id == u) {
+                                              return true;
+                                          }
+                                          return false;
+                                      }),
+                       partners.end());
+        min = Node(none, 0);
+    }
+
+    void sort() {
+        std::sort(partners.begin(), partners.end(), [](const Node &u, const Node &v) {
+            return (u.weight > v.weight || (u.weight == v.weight && u.id < v.id));
+        });
+    }
+};
+
+/**
+ * @ingroup matching
+ * B-Suitor matching finding algorithm.
+ */
+class BSuitorMatcher : public BMatcher {
+public:
+    /**
+     * Computes a 1/2-approximate maximum weight b-matching of an undirected weighted Graph @c G
+     * using the sequential b-Suitor algorithm published by Khan et al. in "Efficient Approximation
+     * Algorithms For Weighted B-Matching", SIAM Journal on Scientific Computing, Vol. 38, Iss. 5
+     * (2016).
+     *
+     * @param G An undirected graph.
+     * @param b A vector of @a b values that represents the max number of edges per vertex @a v in
+     * the b-Matching (b.at(v)).
+     */
+    BSuitorMatcher(const Graph &G, const std::vector<count> &b);
+
+    /**
+     * @param G An undirected graph.
+     * @param b A value @a b that represents the max number of edges per vertex in the b-Matching.
+     * Defaults to the ordinary 1-Matching.
+     */
+    BSuitorMatcher(const Graph &G, count b = 1);
+
+    /**
+     * @param G  An undirected graph.
+     * @param path  A path to a file containing @a b values that represents the max number of edges
+     * per vertex in the b-Matching.
+     */
+    BSuitorMatcher(const Graph &G, const std::string &path);
+
+    ~BSuitorMatcher() override = default;
+
+    /**
+     * Runs the algorithm.
+     */
+    void run() override;
+
+    void buildBMatching();
+
+protected:
+    std::vector<std::unique_ptr<NodeMatchesInfo>> Suitors;
+    std::vector<std::unique_ptr<NodeMatchesInfo>> Proposed;
```

Comment:
We should keep camelCase for member variables.
```suggestion
    std::vector<std::unique_ptr<NodeMatchesInfo>> suitors;
    std::vector<std::unique_ptr<NodeMatchesInfo>> proposed;
```